### PR TITLE
Add Rules unit tests 

### PR DESCRIPTION
**Pull request by shrutip90**
Added Rules unit tests and solutions rules to pass the tests

### CODE REVIEW

File: web/firestore-solution.rules [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      allow create, update: if exists(/databases/(default)/documents/chatrooms/(chatroom)/members/$(request.auth.uid))
+                    && request.resource.data.name == request.auth.token.name 
+                    && (request.resource.data.text is string
+                      && request.resource.data.text.size() <= 300
+                      || request.resource.data.imageUrl is string
+                      && request.resource.data.imageUrl.matches('https?://.*'));
+      allow delete: if false;
+    }
+    
+    // FCM Tokens:
```
Comment (yuchenshi) [inline web/firestore-solution.rules:37]: remove the fcm blocks

**Review by yuchenshi — APPROVED** [review]

File: web/storage-solution.rules [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// Returns true if the uploaded file is an image and its size is below the given number of MB.
+function isImageBelowMaxSize(maxSizeMB) {
+  return request.resource.size < maxSizeMB * 1024 * 1024
+      && request.resource.contentType.matches('image/.*');
+}
+
+service firebase.storage {
+  match /b/{bucket}/o {
+    match /{userId}/{messageId}/{fileName} {
```
Comment (yuchenshi) [inline web/storage-solution.rules:9]: Please consolidate and migrate this into the ideal path you want (in the tests) and also update the index.js code

File: web/firestore-solution.rules (hunk 1 above)
Comment (shrutip90) [reply web/firestore-solution.rules]: Done

File: web/storage-solution.rules (hunk 2 above)
Comment (shrutip90) [reply web/storage-solution.rules]: Done

**Pull request merged**